JustSaying/Messaging/PublishEnvelope.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Collections.Generic;
+using JustSaying.Models;
+
+namespace JustSaying.Messaging
+{
+    public class PublishEnvelope
+    {
+        public PublishEnvelope(Message message)
+        {
+            Message = message ?? throw new ArgumentNullException(nameof(message));
+        }
+
+        public Message Message { get; }
+
+        public TimeSpan? Delay { get; set; }
+
+        public IDictionary<string, MessageAttributeValue> MessageAttributes { get; } = new Dictionary<string, MessageAttributeValue>();
```

Comment:
`MessageAttributeValue` was a straight copy from AWS, now would be a good time to change it if we aren't completely happy with it.

currently its:
```csharp
public class MessageAttributeValue
{
    public IReadOnlyCollection<byte> BinaryValue { get; set; }

    public string StringValue { get; set; }

    public string DataType { get; set; }
}
```